xbmc/addons/Skin.cpp

```diff
@@ -369,6 +369,8 @@ const INFO::CSkinVariableString* CSkinInfo::CreateSkinVariable(const std::string
 
 void CSkinInfo::OnPreInstall()
 {
+  if (IsInUse())
+    CApplicationMessenger::GetInstance().SendMsg(TMSG_EXECUTE_BUILT_IN, -1, -1, nullptr, "UnloadSkin");
```

Comment:
I would highly, highly welcome it - if we could encapsulate that into windows specific code. Any ideas?